### ISSUE TITLE
"indirect call to null" in `_ts_parser_parse_wasm`

### ISSUE THREAD

**Issue by CyberShadow** [tree-sitter/tree-sitter#1179]
If I try to build my grammar with `tree-sitter build-wasm --docker`, it successfully creates a `.wasm` file. However, it appears to be broken.

Trying to use it by adding it to [the playground](https://tree-sitter.github.io/tree-sitter/playground) results in the following error:

```
Uncaught (in promise) RuntimeError: indirect call to null
    _ts_parser_parse_wasm https://cy.md/d/tree-sitter-d/tree-sitter.js:1
    parse https://cy.md/d/tree-sitter-d/tree-sitter.js:1
    handleCodeChange https://tree-sitter.github.io/tree-sitter/assets/js/playground.js:115
    handleLanguageChange https://tree-sitter.github.io/tree-sitter/assets/js/playground.js:101
    async* https://tree-sitter.github.io/tree-sitter/assets/js/playground.js:78
    async* https://tree-sitter.github.io/tree-sitter/assets/js/playground.js:461
tree-sitter.wasm:111437:1
```

If I use a .wasm file for an existing grammar downloaded from https://tree-sitter.github.io/ , that works. So, something is going wrong in the build process causing a broken .wasm file to be created.

Demo: https://cy.md/d/tree-sitter-d/
Grammar: https://github.com/cybershadow/tree-sitter-d/tree/70231be2888d49c9249b8503505ecca9367d1888

**Comment by CyberShadow** [comment]
In Chromium I see a different error:

```
Uncaught (in promise) RuntimeError: function signature mismatch
    at tree-sitter.wasm:0x1b34d
    at tree-sitter.wasm:0x27141
    at ts_parser_parse_wasm (tree-sitter.wasm:0x2eeb8)
    at Object.Module._ts_parser_parse_wasm (tree-sitter.js:1)
    at Parser.parse (tree-sitter.js:1)
    at handleCodeChange (playground.js:115)
    at handleLanguageChange (playground.js:101)
    at async playground.js:78
```

This seems to suggest that there may be a version incompatibility? However, I have 0.19.5 in `package.json` and `package-lock.json`, and I downloaded `tree-sitter.js` and `tree-sitter.wasm` from [the corresponding GitHub release](https://github.com/tree-sitter/tree-sitter/releases/tag/v0.19.5).

**Comment by ahlinc** [comment]
@CyberShadow FYI, I've described my investigation for the problem here: https://github.com/tree-sitter/tree-sitter-haskell/issues/35#issuecomment-857387920

**Comment by CyberShadow** [comment]
@ahlinc Thanks for pointing me in the right direction. I submitted #1180 which should fix this permanently.

**Comment by ahlinc** [comment]
@maxbrunsfeld this was fixed by #1180 and can be closed.
